src/main/java/com/hackathonorganizer/hackathonwriteservice/team/service/TeamService.java

```diff
@@ -0,0 +1,93 @@
+package com.hackathonorganizer.hackathonwriteservice.team.service;
+
+import com.hackathonorganizer.hackathonwriteservice.hackathon.model.Hackathon;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.repository.HackathonRepository;
+import com.hackathonorganizer.hackathonwriteservice.team.exception.ResourceNotFoundException;
+import com.hackathonorganizer.hackathonwriteservice.team.model.Tag;
+import com.hackathonorganizer.hackathonwriteservice.team.model.Team;
+import com.hackathonorganizer.hackathonwriteservice.team.model.dto.TagRequest;
+import com.hackathonorganizer.hackathonwriteservice.team.model.dto.TeamRequest;
+import com.hackathonorganizer.hackathonwriteservice.team.model.dto.TeamResponse;
+import com.hackathonorganizer.hackathonwriteservice.team.repository.TeamRepository;
+import com.hackathonorganizer.hackathonwriteservice.team.utils.TeamMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TeamService {
+
+
+    private final TeamRepository teamRepository;
+    private final TagService tagService;
+    //TODO change to service when it will be ready
+    private final HackathonRepository hackathonRepository;
+
+    public TeamResponse create(TeamRequest teamRequest) {
+        val teamToSave = Team.builder()
+                .ownerId(teamRequest.ownerId())
+                .hackathon(getHackathon(teamRequest.hackathonId()))
+                .teamMembersIds(teamRequest.teamMembersIds())
+                .tags(getTags(teamRequest.tags()))
+                .build();
+        log.info("Trying to save new team {}", teamToSave);
+        return TeamMapper.toResponse(teamRepository.save(teamToSave));
+    }
+
+    public TeamResponse editById(Long id, TeamRequest teamRequest) {
+        return teamRepository.findById(id)
+                .map(teamToEdit -> {
+                    teamToEdit.setOwnerId(teamRequest.ownerId());
+                    teamToEdit.setHackathon(getHackathon(teamRequest.hackathonId()));
+                    teamToEdit.setTeamMembersIds(teamRequest.teamMembersIds());
+                    teamToEdit.setTags(getTags(teamRequest.tags()));
+                    return TeamMapper.toResponse(teamRepository.save(teamToEdit));
+                }).orElseThrow(() -> {
+                    log.error(String.format("Team id = %d not found", id));
+                    return new ResourceNotFoundException(String.format("Team id = %d not found", id));
+                });
+    }
+
+    public TeamResponse editPartialById(Long id, TeamRequest teamRequest) {
+        return teamRepository.findById(id)
+                .map(teamToEdit -> {
```

Comment:
To samo tu, tez mozna wydzielic kod do mappera i wywolywac referencje do metody

Reply:
Map nie ma nic wspólnego z mapperem

Reply:
Do prywatnej tak jak pisalem wczesniej

Reply:
Ale co to da i co mam wynieść? Nie rozumiem o co chodzi

Reply:
Ja tu nic nie mapuje

Reply:
Zgodnie z clean codem, funkcja powinna byc jak najmniejsza, dlatego wydziela sie kod do prywatnych funkcji

Reply:
No ale co mam wydzielić jak tu jest logika cała? Tu nie ma co wydzielac

Reply:
Ja bym cala ta lambde przeniosl do prywatnej metody

Reply:
I jak ją nazwać?

Reply:
np. updateByTeamRequest, wedlug mnie poprawi to czytelnosc

Reply:
Dla mnie to jest bezsensu bo po to jest metoda edit żeby w niej była logika do edytowania. Rozumiem wynieść inne metody które nie mają bezpośrednio wpływu na logikę ale przy tym nie widzę sensu. Jak się większość osób wypowie żeby przenieść do prywatnych metod to przeniosę